cookbook/src/tx_segwit-v0.md

```diff
@@ -0,0 +1,342 @@
+   This is possible because `Witness` implements the [`Default`](https://doc.rust-lang.org/std/default/trait.Default.html) trait.
+
+In `let spend = TxOut {...}` we are instantiating the spend output.
+Inside the [`TxOut`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/transaction/struct.TxOut.html) struct we are setting the following fields:
+
+- `value` is the amount we are spending; it is a [`u64`](https://doc.rust-lang.org/std/primitive.u64.html) type.
+   We are using the `const SPEND_AMOUNT` that we defined earlier.
+- `script_pubkey` is the script code required to spend a P2WPKH output; it is a [`ScriptBuf`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/script/struct.ScriptBuf.html) type.
+   We are using the [`script_pubkey`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/address/struct.Address.html#method.script_pubkey) method to generate the script code.
+   This will lock the output to the receiver's address.
+
+In `let change = TxOut {...}` we are instantiating the change output.
+It is very similar to the `spend` output, but we are now using the `const CHANGE_AMOUNT` that we defined earlier.
+And also we are locking the output to an address that we control:
+the `wpkh` public key hash that we generated earlier.
+This is done by setting the `script_pubkey` field to [`ScriptBuf::new_v0_p2wpkh(&wpkh)`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/script/struct.ScriptBuf.html#method.new_v0_p2wpkh),
+which generates P2WPKH-type of script pubkey.
+
+In `let unsigned_tx = Transaction {...}` we are instantiating the transaction we want to sign and broadcast using the [`Transaction`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/transaction/struct.Transaction.html) struct.
+We set the following fields:
+
+- `version` is the transaction version; it is a [`i32`](https://doc.rust-lang.org/std/primitive.u32.html) type.
+   We are using version `2` which means that [BIP68](https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki) applies.
+- `lock_time` is the transaction lock time;
+   it is a [`LockTime`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/locktime/absolute/enum.LockTime.html) enum.
+   We are using the [`from_height`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/locktime/absolute/enum.LockTime.html#method.from_height) method to set the lock time to the height `0`.
+   This will make the transaction valid immediately.
+- `input` is the input vector; it is a [`Vec<TxIn>`](https://doc.rust-lang.org/std/vec/struct.Vec.html) type.
+   We are using the `input` variable that we defined earlier wrapped in the [`vec!`](https://doc.rust-lang.org/std/macro.vec.html) macro for convenient initialization.
+- `output` is the output vector; it is a [`Vec<TxOut>`](https://doc.rust-lang.org/std/vec/struct.Vec.html) type.
+   We are using the `spend` and `change` variables that we defined earlier wrapped in the [`vec!`](https://doc.rust-lang.org/std/macro.vec.html) macro for convenient initialization.
```

Comment:
Perhaps mention that order does not matter but by convention it goes `[spend, change]` (I think that is the convention anyways).

Reply:
I didn't understand what you meant

Reply:
In the code the outputs vector is `[spend, change]` but could have been `[change, spend]`. Both are correct (I believe), when I wrote this I spent a little time looking up if there was a convention on this and it seems that `[spend, change]` is it but I'm not 100% sure (I could be mixing it up with the fact that in derivation path 0 is external and 1 is change) .